core/src/main/scala-3/scairdl/IRElements.scala

```diff
@@ -223,59 +224,85 @@ case class OperationDef(
   }
 
   def Generateparsefunction(format: Seq[FormatDirective]): String = {
-    val operandVars = format.collect { case OperandDirective(name) => name }
+    val patternMappings = format.zipWithIndex.map {
+      case (OperandDirective(name), i) =>
+        (s"${name}", "Parser.ValueUse", "operand")
 
-    val typeVars = format.collect {
-      case TypeDirective(OperandDirective(name)) => name
-    }
+      case (TypeDirective(OperandDirective(name)), i) =>
+        (s"type_${name}", "parser.Type", "operandType")
 
-    val resultVars = format.collect { case ResultTypeDirective(name) => name }
+      case (ResultTypeDirective(name), i) =>
+        (s"type_${name}", "parser.Type", "resultType")
 
-    val operandPatternVars = operandVars.zipWithIndex.map { case (n, i) =>
-      s"${n}_$i"
+      case (LiteralDirective(lit), i) =>
+        (s"lit_$i", s""""$lit"""", "literal")
     }
-    val typePatternVars = typeVars.zipWithIndex.map { case (n, i) =>
-      s"type_${n}_$i"
+
+    val patternVariables = patternMappings
+      .collect {
+        case (variable, _, directiveType) if directiveType != "literal" =>
+          variable
+      }
+      .mkString(", ")
+    // this part is messy. I'll improve it
```

Comment:
Could either remove this or write a `TODO` here as a hint in case someone else wants to improve it later?